### PR TITLE
Accept valacdos files

### PR DESCRIPTION
**Pull request by pauldambra**
Allows the checker to be instantiated with the contents of arbitrary valacdos files.

### CODE REVIEW

**Comment by pauldambra** [comment]
@rokeyge would this work for you?

File: README.md [hunk 1]
```diff
@@ -59,6 +59,31 @@ Assert.AreEqual(true, outcome.Result);
...
+#### Valacdos File Contents
+
+The library is built with a fixed version of the valacdos files (see below).
+
+The valacdos files can be provided to the modulus checker constructor. 
+
+```
+const string weightMappingsContent = "blah blah";
+const string scsubtabContent = "blah blah blah";
+var modulusChecker = new ModulusChecker(weightMappingsContent, scsubtabContent);
```
Comment (rokeyge) [inline README.md:71]: :+1:

File: .semver [hunk 2]
```diff
@@ -1 +1 @@
-2.3.0
+2.4.0
```
Comment (rokeyge) [inline .semver:1]: minor: major change for the library?

File: ModulusChecking/Loaders/SortCodeSubstitution.cs [hunk 3]
```diff
@@ -8,13 +8,11 @@ namespace ModulusChecking.Loaders
...
-        private Dictionary<string, string> _sortCodeSubstitutionSource;
+        private static Dictionary<string, string> _sortCodeSubstitutionSource;
 
-        private void SetupDictionary()
+        public SortCodeSubstitution(string scsubtabFileContents)
         {
-            if (_sortCodeSubstitutionSource != null) return;
-
-            _sortCodeSubstitutionSource = Resources.scsubtab
+            _sortCodeSubstitutionSource = scsubtabFileContents
```
Comment (rokeyge) [inline ModulusChecking/Loaders/SortCodeSubstitution.cs]: Maybe simple null checking here?

**Comment by rokeyge** [comment]
Do you want to add some tests on the high level `new ModulusChecker(weightMappingFileContents, scsubtabFileContents)`?

**Comment by rokeyge** [comment]
Changes make sense to me! Thanks!

File: .semver (hunk 2 above)
Comment (pauldambra) [reply .semver:1]: Yeah, so semver for me is breaking:feature:fix

We've added a feature with breaking existing behaviour (or believe we have :)) so bump the feature version

File: ModulusChecking/Loaders/SortCodeSubstitution.cs (hunk 3 above)
Comment (pauldambra) [reply ModulusChecking/Loaders/SortCodeSubstitution.cs]: good spot! potentially user provided content now and we're giving poor feedback if it's obviously wrong

**Pull request merged**